### ISSUE TITLE
deepmd-kit 2.0.0 multi gpu training only one GPU used

### ISSUE THREAD

**Issue by tonystarkiss** [deepmodeling/deepmd-kit#1183]
**Summary**
I used deepmd-kit 2.0.0 multi gpu training, there are 4 CPU processors,  but only one GPU is used,
![image](https://user-images.githubusercontent.com/30335530/135715353-4a73e2e7-ea0d-4a1d-af87-1fce49aa3df1.png)
![image](https://user-images.githubusercontent.com/30335530/135715374-561c8886-e161-42bf-a2ec-e3e5379c5b73.png)
![image](https://user-images.githubusercontent.com/30335530/135715390-9281c730-a6e2-4b5e-9ca2-a385b5039db5.png)

**Deepmd-kit version, installation way, input file, running commands, error log, etc.**
the deepmd-kit was pulled from :
docker pull ghcr.io/deepmodeling/deepmd-kit:2.0.0_cuda10.1_gpu
or use offline packages
both have the same bug.

the input file:
deepmd-kit/examples/water/se_e2_a

**Steps to Reproduce**
CUDA_VISIBLE_DEVICES=4,5,6,7 horovodrun -np 4 /data/deepmd-kit/bin/dp train input.json 
or CUDA_VISIBLE_DEVICES=4,5,6,7 mpirun -np 4 /data/deepmd-kit/bin/dp train input.json

**Further Information, Files, and Links**

**Comment by shishaochen** [comment]
@tonystarkiss Pleast attach your full log before training iterations.

**Comment by njzjz** [comment]
@liudfgoo Is your horovod built against your MPI program?

**Comment by liudfgoo** [comment]
@njzjz I think it is but I'm not sure. How to confirm that? And the below is the result of the command `horovodrun -cb`
<img width="351" alt="d0ec4e4410b0b303858550d783fdb56" src="https://user-images.githubusercontent.com/49781630/147896041-54c5708a-3aaa-4774-a3d9-c24828dc925a.png">

**Comment by njzjz** [comment]
@liudfgoo Horovod v0.22.1 has a known issue in #1090. If you are using MPICH, you need to upgrade Horovod to v0.23.0.

**Comment by liudfgoo** [comment]
@njzjz Thanks for your help! According to your suggestion, I have solved this problem. When I used `mpirun -np 4  dp train input.json `, the four gpu cards could run successfully. 
<img width="611" alt="50b80b97fd5b9bb55de3ca9d992c0d6" src="https://user-images.githubusercontent.com/49781630/148234409-f28712b1-478f-43a6-8d64-3bb1cb60848d.png">

However, maybe due to the small data, the running speed with 4 gpu cards was lower than 1 gpu card.  After that, I tried to use 2 gpu cards, which also had a lower speed than one card. Perhaps the advantages of multiple cards will become apparent only with larger data sets.
